File: spring-data-jpa-querydsl/jpa-basic/src/main/java/me/kzv/jpabasic/domain/member/Member.java
package me.kzv.jpabasic.domain.member;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import me.kzv.jpabasic.domain.team.Team;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Entity
@Table(name = "member")
public class Member {

    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    private Long id;

    private String memberName;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "team_id") // 실제 db에 물리적으로 fk를 건다.
    private Team team;
}

File: spring-data-jpa-querydsl/jpa-best-practices/src/main/kotlin/me/kzv/jpabestpractices/domain/member/Member.kt
package me.kzv.jpabestpractices.domain.member

import jakarta.persistence.*
import me.kzv.jpabestpractices.domain.team.Team
import me.kzv.jpabestpractices.supports.AuditableEntity
import org.hibernate.annotations.Where

@Entity
@Table(name = "tbl_member")
@Where(clause = "status = 'ACTIVE'")
class Member(
    @JoinColumn(
        name = "team_id",
        foreignKey = ForeignKey(name = "fk_team_id"),
        nullable = false,
    )
    @ManyToOne(fetch = FetchType.LAZY)
    val team: Team,

    val name: String,

    @Enumerated(EnumType.STRING)
    var status: MemberStatus = MemberStatus.ACTIVE
) : AuditableEntity() {

}